pykrige/core.py

```diff
@@ -54,6 +58,26 @@
 from scipy.optimize import minimize
 
 
+def great_circle_distance(lon1, lat1, lon2, lat2):
+    # Calculate dot product of both vectors:
+    lat1 *= np.pi/180.0
+    lat2 *= np.pi/180.0
+    d = np.cos((lon1-lon2)*np.pi/180.0)*np.cos(lat1)*np.cos(lat2) \
+        + np.sin(lat1)*np.sin(lat2)
+    # Angle is arccos of dot product. Avoid errors caused
+    # by numerics:
+    d[d>1.0] = 1.0
+    d[d<-1.0]=-1.0
```

Comment:
Yes, NAN is returned by arccos. Some small numerical errors might cause |d|>1.0. even though it should not be possible analytically. I believe I've encountered this problem in similar code I used in another setting, so I decided to add it here as well.